src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -94,9 +106,17 @@ public void CompileMethod(MethodCodeNode methodCodeNodeNeedingCode)
                     }
                 }
 
+                IntPtr exception;
                 IntPtr nativeEntry;
                 uint codeSize;
-                _compile(_jit, _comp, ref methodInfo, flags, out nativeEntry, out codeSize);
+                JitWrapper(out exception, _jit, _comp, ref methodInfo, flags, out nativeEntry, out codeSize);
+                if (exception != IntPtr.Zero)
+                {
+                    char* szMessage = GetExceptionMessage(exception);
+                    string message = szMessage != null ? new string(szMessage) : "JIT Exception";
+                    FreeException(exception);
```

Comment:
The exception can be caught by the JIT too - we will leak memory if the JIT catches the exception (e.g. if the exception happens during inlining - it is not fatal). To avoid the leak, we may want to keep track of all exceptions that were thrown during JITing and free them in FlushPins (this method can use a better name...).